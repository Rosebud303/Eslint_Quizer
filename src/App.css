h1 {
    height: 130px;
    text-align: center;
    padding-top: 40px;
}

body {
    background: #9b4dbb;
}

.score {
    text-align: center;
}

.button-area {
    display: flex;
    justify-content: center;
}

.startQuiz, .show-wrong{
    margin: 0;
}
