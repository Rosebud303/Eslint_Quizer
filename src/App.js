import React, { Component } from 'react';
import QuizContainer from './QuizContainer.js';
import RuleLibrary from './RuleLibrary.js';
import './App.css';
import WrongContainer from './WrongContainer.js'

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      categories: [],
      questions: [],
      showQuiz: false,
      incorrectAnswers: [],
      showWrong: false
    }
  };

  componentDidMount() {
    let wrongStorage = JSON.parse(localStorage.getItem('wrongAnswers'));
    let updatedCurrentWrong = [];
    fetch('https://fe-apps.herokuapp.com/api/v1/memoize/1901/duydata/categories')
    .then(response => response.json())
    .then(data => {
      this.setState({
        categories: data.categories,
      })
    })
    .catch(error => {
      this.setState({error: error.message})
    });
    fetch('https://fe-apps.herokuapp.com/api/v1/memoize/1901/duydata/quizQuestions')
    .then(response => response.json())
    .then(data => {
      this.setState({
        questions: data.quizQuestions,
      })
    })
    .catch(error => {
      this.setState({error: error.message})
    });
    if(wrongStorage !== null) {
      updatedCurrentWrong = wrongStorage
    }
    this.setState({ incorrectAnswers: updatedCurrentWrong});
  }

  toggleWrong = () => {
    this.setState({showWrong: !this.state.showWrong});
  };

  toggleQuiz = () => {
    this.setState({showQuiz: !this.state.showQuiz});
  };

  render() {
    const { categories, questions, showQuiz, incorrectAnswers, showWrong } = this.state;
    const { toggleQuiz, toggleWrong } = this;
    return (
      <div className="App">
        <h1>memoize.eslint!</h1>
        <h3 className='score'>Score: {incorrectAnswers.length}/30</h3>
        <RuleLibrary categories={categories}/>
        { showWrong &&
            <div>
              <WrongContainer questions={questions}
                              incorrects={incorrectAnswers}
                              toggle={toggleWrong}
              />
            </div>
        }
        <div className='button-area'>
          <button className='startQuiz' 
                  onClick={toggleQuiz}>
                  Take Quiz!
          </button>
          <button className='show-wrong'
                  onClick={toggleWrong}>
                  Show Missed...
          </button>
        </div>
        { showQuiz &&
            <QuizContainer questions={questions}
                           toggle={toggleQuiz}
                           wrong={incorrectAnswers}
                           categories={categories}
            />
        }
      </div>
    )
  };
};

export default App;
