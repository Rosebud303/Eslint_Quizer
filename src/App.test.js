import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme'

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App />
    )
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should have default state', () => {
    expect(wrapper.state()).toEqual({categories: [],
                                    questions: [],
                                    showQuiz: false,
                                    incorrectAnswers: [],
                                    showWrong: false
                                    });
  });

  it('Should toggle the state of showQuiz', () => {
    expect(wrapper.state('showQuiz')).toEqual(false);
    wrapper.instance().toggleQuiz();
    expect(wrapper.state('showQuiz')).toEqual(true);
  });

  it('Should toggle the state of showWrong', () => {
    expect(wrapper.state('showWrong')).toEqual(false);
    wrapper.instance().toggleWrong();
    expect(wrapper.state('showWrong')).toEqual(true);
  });
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
