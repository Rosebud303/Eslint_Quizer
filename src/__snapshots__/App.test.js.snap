// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match the snapshot 1`] = `
<div
  className="App"
>
  <h1>
    memoize.eslint!
  </h1>
  <h3
    className="score"
  >
    Score: 
    0
    /30
  </h3>
  <RuleLibrary
    categories={Array []}
  />
  <div
    className="button-area"
  >
    <button
      className="startQuiz"
      onClick={[Function]}
    >
      Take Quiz!
    </button>
    <button
      className="show-wrong"
      onClick={[Function]}
    >
      Show Missed...
    </button>
  </div>
</div>
`;
